dart/lib/src/browser_client.dart

```diff
@@ -41,19 +42,19 @@ class SentryBrowserClient extends SentryClient {
     );
   }
 
-  SentryBrowserClient._(SentryOptions options, {String origin, String platform})
+  SentryBrowserClient._(SentryOptions options,
+      {String origin, @required String platform})
       : super.base(
           options,
-          origin: origin,
-          sdk: Sdk(name: browserSdkName, version: sdkVersion),
-          platform: platform,
+          transport: Transport(
+            dsn: options.dsn,
+            httpClient: options.httpClient,
+            clock: options.clock,
+            compressPayload: false,
+            sdk: Sdk(name: browserSdkName, version: sdkVersion),
+            headersBuilder: SentryClient.buildHeaders,
```

Comment:
this could be done inside of the Transport layer I guess, why does the client need to know about the headers?

Reply:
updated  : now it uses a conditional platform in transport.